addons/product/product.py

```diff
@@ -746,7 +746,10 @@ def _get_uom_id(self, cr, uid, *args):
     def _check_uom(self, cr, uid, ids, context=None):
         for supplier_info in self.browse(cr, uid, ids, context=context):
             if supplier_info.product_uom and supplier_info.product_uom.category_id.id <> supplier_info.product_id.uom_id.category_id.id:
-                return False
```

Comment:
OK, I think a better fix is possible and it's out of scope for this
change so please remove.